src/policy/mallocspace/global.rs

```diff
@@ -152,16 +152,19 @@ impl<VM: VMBinding> MallocSpace<VM> {
         }
     }
 
-    pub fn alloc(&self, tls: VMThread, size: usize) -> Address {
+    pub fn alloc(&self, tls: VMThread, align: usize, size: usize) -> Address {
         // TODO: Should refactor this and Space.acquire()
         if VM::VMActivePlan::global().poll(false, self) {
             assert!(VM::VMActivePlan::is_mutator(tls), "Polling in GC worker");
             VM::VMCollection::block_for_gc(VMMutatorThread(tls));
             return unsafe { Address::zero() };
         }
 
-        let raw = unsafe { calloc(1, size) };
+        let raw = unsafe { aligned_alloc(align, size) };
         let address = Address::from_mut_ptr(raw);
+        // zero
+        // malloc+memset is always slower than calloc. But to support alignment, we do not have a better choice.
+        crate::util::memory::zero(address, size);
```

Comment:
Should we not check if the address is not a null ptr before zeroing it? I worry that this might segfault otherwise.

Reply:
We should do a null check. Thanks for pointing this out.